test/CMakeLists.txt

```diff
@@ -209,7 +209,7 @@ foreach(SDK ${SWIFT_SDKS})
     if("${SDK}" STREQUAL "IOS" OR "${SDK}" STREQUAL "TVOS" OR "${SDK}" STREQUAL "WATCHOS")
       # These are supported testing SDKs, but their implementation of
       # `command_upload_stdlib` is hidden.
-    elseif("${SDK}" STREQUAL "ANDROID")
+    elseif("${SDK}" STREQUAL "ANDROID" AND NOT "${SWIFT_HOST_VARIANT}" STREQUAL "android")
```

Comment:
This seems weird ... the SDK that you are compiling against is the correct check.

Reply:
Yes, I had to check either `CMAKE_SYSTEM_NAME` or `SWIFT_HOST_VARIANT` to know whether the tests are being compiled _on_ Android, as described above, so I could skip this adb configuration meant for remotely testing on other Android devices.

Reply:
This is the one place where I kept a SWIFT_HOST_VARIANT check in my latest commit, because I couldn't think of anything better. Ideally, this adb connection should be disabled in Termux, enabled when cross-compiling the Swift stdlib from linux ie the current setup, and optionally enabled when cross-compiling both the Swift host tools, ie the bin/swift compiler, and the stdlib for Android. I couldn't think of a good way to do the last using existing variables, so this simply disables using adb for the first and third situations.

I'll add a comment like you said.